### PR TITLE
Create CRUD for analyses

### PR DESCRIPTION
**Pull request by nikoleta01**

### CODE REVIEW

File: src/router/index.ts [hunk 1]
```diff
@@ -71,6 +71,89 @@ const routes: Array<RouteRecordRaw> = [
 				]
 
 
+			},
+			// {
```
Comment (MCFreddie777) [inline src/router/index.ts]: Unnecessary comment.

File: src/router/index.ts [hunk 2]
```diff
@@ -71,6 +71,89 @@ const routes: Array<RouteRecordRaw> = [
...
+						name: 'analysis-edit',
+						meta: {
+							title: 'Upraviť laboratórium'
+						},
+						component: () =>
+							import(
+								/* webpackChunkName: "AnalysisEdit" */ '@/views/Analysis/AnalysisEdit.vue'
+							)
+					},
+					// {
```
Comment (MCFreddie777) [inline src/router/index.ts]: Comment here as well. Please remove it.

File: src/store/user/user.types.ts [hunk 3]
```diff
@@ -3,4 +3,9 @@ export interface User {
     first_name: string;
     last_name: string;
     email: string;
-}
\ No newline at end of file
+}
+
+export interface State {
+	analyses: User[];
```
Comment (MCFreddie777) [inline src/store/user/user.types.ts]:
```typescript
users: User[];
user: User;
```

File: src/views/Analysis/AnalysisCreate.vue [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+	},
+	methods: {
+		...mapActions('AnalysisStore', ['saveAnalysis', 'resetState']),
+		...mapActions('AppStore', ['setAlert']),
+		...mapActions('LabStore', ['fetchLabs']),
+		...mapActions('ToolStore', ['fetchTools']),
+		...mapActions('SampleStore', ['fetchSamples']),
+
+		addTool: function () {
+			console.log(this.analysis);
```
Comment (MCFreddie777) [inline src/views/Analysis/AnalysisCreate.vue]: `console.log`

File: src/views/Analysis/AnalysisCreate.vue [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+			if (this.analysis) {
```
Comment (MCFreddie777) [inline src/views/Analysis/AnalysisCreate.vue]: Unnecessary check. This analysis is always present, as it it in the `data` function.

File: src/views/Analysis/AnalysisCreate.vue [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+			} else {
+				this.saving = false;
+				(this.$refs.form as HTMLFormElement).reportValidity();
+			}
+		}
+	},
+	mounted: function () {
+		// initial loader
+		// fetch data from BE
+		this.fetchLabs();
```
Comment (MCFreddie777) [inline src/views/Analysis/AnalysisCreate.vue]: Idea: `Promise.all` and show loader maybe? (as all are promises)

File: src/views/Analysis/AnalysisEdit.vue [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+		...mapActions('AppStore', ['setAlert']),
+		...mapActions('LabStore', ['fetchLabs']),
+		...mapActions('ToolStore', ['fetchTools']),
+		...mapActions('SampleStore', ['fetchSamples']),
+
+		addTool: function () {
+			this.analysis.tools.push({});
+		},
+		deleteTool: function (index: number) {
+			this.analysis.tools = this.analysis.tools.splice(index, 1); // 2nd parameter means remove one item only
```
Comment (MCFreddie777) [inline src/views/Analysis/AnalysisEdit.vue]: Uneccessary coment, everybody knows what `Array.splice` does.

File: src/views/Analysis/AnalysisEdit.vue [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+				this.saving = false;
+				(this.$refs.form as HTMLFormElement).reportValidity();
+			}
+		}
+	},
+	mounted: function () {
+		// initial loader
+		this.loading = true;
+		// fetch data from BE
+		this.fetchAnalysis(this.$route.params.id).finally(() => {
```
Comment (MCFreddie777) [inline src/views/Analysis/AnalysisEdit.vue]: So, only when analysis is fetched, the loader is hidden? What about the other async assets as fetchlabs or fetchTools? 

Please use `Promise.all()` and then disable the loader.

**Review by MCFreddie777 — CHANGES_REQUESTED** [review]
Good job, left some comments.

File: src/store/user/user.types.ts (hunk 3 above)
Comment (nikoleta01) [reply src/store/user/user.types.ts]: good catch

**Review by MCFreddie777 — APPROVED** [review]

**Pull request merged**